packages/app/src/lib/client/marinade.ts

```diff
@@ -0,0 +1,245 @@
+    config,
+    program,
+    epochInfo.epoch
+  );
+  const previousManagementAccount =
+    await getOrderUnstakeTicketManagementAccount(
+      config,
+      program,
+      epochInfo.epoch - 1
+    );
+
+  // TODO Add instruction to close an arbitrary ticket, in case an epoch gets missed
+
+  const previousEpochTickets = previousManagementAccount.account
+    ? await findAllTickets(
+        program.provider.connection,
+        config,
+        previousManagementAccount.account
+      )
+    : [];
+  const previousEpochTicketAccountMetas = previousEpochTickets.map(
+    (ticket) => ({
+      pubkey: ticket,
+      isSigner: false,
+      isWritable: true,
+    })
+  );
+  // TODO add check to see if rebalancing is needed
+
+  // TODO Split rebalancing (order unstake) and claiming tickets - claiming tickets can be one instruction each
+  // no need for remaining accounts etc.
+  // Then the client bundles them together. However, this means the client will not be guaranteed to do a rebalancing
+  // which costs a bit of rent.
+
+  // TODO incrementing on the client side like this will cause clashes in future, we need to replace it
+  const index = (managementAccount?.account?.tickets.toNumber() ?? 0) + 1;
+  const [orderUnstakeTicketAccount, orderUnstakeTicketAccountBump] =
+    findOrderUnstakeTicketAccount(
+      config,
+      BigInt(epochInfo.epoch),
+      BigInt(index)
+    );
+
+  type Accounts = Parameters<
+    ReturnType<typeof program.methods.triggerPoolRebalance>["accounts"]
+  >[0];
+
+  const accounts: Accounts = {
+    state: stateAddress,
+    marinadeState: marinadeState.marinadeStateAddress,
+    gsolMint: sunriseStakeState.gsolMint,
+    msolMint: marinadeState.mSolMint.address,
+    liqPoolMint: marinadeState.lpMint.address,
+    liqPoolSolLegPda: await marinadeState.solLeg(),
+    liqPoolMsolLeg: marinadeState.mSolLeg,
+    liqPoolMsolLegAuthority: await marinadeState.mSolLegAuthority(),
+    liqPoolMintAuthority: await marinadeState.lpMintAuthority(),
+    liqPoolTokenAccount: liqPoolAssociatedTokenAccountAddress,
+    reservePda: await marinadeState.reserveAddress(),
+    treasuryMsolAccount: marinadeState.treasuryMsolAccount,
+    getMsolFrom: msolAssociatedTokenAccountAddress,
+    getMsolFromAuthority: msolTokenAccountAuthority,
+    orderUnstakeTicketAccount,
+    orderUnstakeTicketManagementAccount: managementAccount.address,
+    // Utilising the new "optional named accounts" feature in Anchor 0.26.0
+    // to pass this only if it exists.
+    // Since this passes null instead of the address, what is to stop the client from "lying"?
+    // Well, nothing in the general case, but in this case, it is in the caller's interest to pass the account,
+    // as it allows them to claim rent on closure of this account and the tickets.
+    previousOrderUnstakeTicketManagementAccount:
+      previousManagementAccount.account
+        ? previousManagementAccount.address
+        : null,
```

Comment:
I think this should be undefined instead of null. 
Otherwise I got this error:
ERROR in src/lib/client/marinade.ts:217:5
TS2322: Type 'PublicKey | null' is not assignable to type 'Address | undefined'.
  Type 'null' is not assignable to type 'Address | undefined'.

Reply:
Can you check you have updated to anchor 0.26?